backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/RemoveImageCommandCallback.java

```diff
@@ -0,0 +1,82 @@
+package org.ovirt.engine.core.bll.storage.disk.image;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.enterprise.inject.Typed;
+import javax.inject.Inject;
+
+import org.ovirt.engine.core.bll.tasks.CommandCoordinatorUtil;
+import org.ovirt.engine.core.bll.tasks.interfaces.CommandCallback;
+import org.ovirt.engine.core.common.action.RemoveImageParameters;
+import org.ovirt.engine.core.common.businessentities.AsyncTaskStatus;
+import org.ovirt.engine.core.compat.CommandStatus;
+import org.ovirt.engine.core.compat.Guid;
+import org.ovirt.engine.core.vdsbroker.vdsbroker.VDSErrorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Typed(RemoveImageCommandCallback.class)
+public class RemoveImageCommandCallback implements CommandCallback {
+    private static final Logger log = LoggerFactory.getLogger(RemoveImageCommandCallback.class);
+
+    @Inject
+    private CommandCoordinatorUtil commandCoordinatorUtil;
+
+    @Override
+    public void doPolling(Guid cmdId, List<Guid> childCmdIds) {
+        RemoveImageCommand<RemoveImageParameters> command = getRemoveImageCommand(cmdId);
+        Set<Guid> taskIds = new HashSet<>(command.getReturnValue().getVdsmTaskIdList());
+        Map<Guid, AsyncTaskStatus> idToTaskStatusMap = commandCoordinatorUtil.pollTasks(taskIds);
+        for (Map.Entry<Guid, AsyncTaskStatus> idToTaskStatus : idToTaskStatusMap.entrySet()) {
+            Guid id = idToTaskStatus.getKey();
+            AsyncTaskStatus status = idToTaskStatus.getValue();
+            if (status.getTaskIsRunning()) {
+                log.info("Waiting on remove image command to complete the task (taskId = '{}')", id);
+                return;
+            }
+        }
+
+        List<AsyncTaskStatus> failedTasks = new ArrayList<>();
+        idToTaskStatusMap.forEach((id, status) -> {
+            if (!status.getTaskEndedSuccessfully()) {
+                failedTasks.add(status);
+             }
+        });
+        if (failedTasks.isEmpty()) {
+            command.setSucceeded(true);
+            command.setCommandStatus(CommandStatus.SUCCEEDED);
+            command.persistCommand(command.getParameters().getParentCommand());
+            log.info("Remove image command has completed successfully for image '{}' with async task '{}'.",
+                    command.getParameters().getImageId(), taskIds);
+        } else {
+            RuntimeException exception = failedTasks.get(0).getException();
+            if (exception instanceof VDSErrorException) {
+                command.getParameters()
+                        .setAsyncTaskErrorMessage(((VDSErrorException) exception).getVdsError().getMessage());
+            }
+            command.setSucceeded(false);
+            command.setCommandStatus(CommandStatus.FAILED);
+            log.info("Remove image command has failed for images '{}' with async task '{}'.",
```

Comment:
For `task` you print the `List` of task statuses, might be more than 1 task status? If so, need to to update the message? Also you intended to print the statuses or IDs of the tasks?
Also for `images` (plural) you print just a single Image ID?

Reply:
You are right, I have fixed the message to show the disk ID - the task removes image group (disk), which might have some images, but we don't have their IDs in this context, and not worth going to the database only for the sake of additional logging - their IDs will be logged in the command itself afterwards

Reply:
As for ID or Status - I think printing status here will be somewhat more informative and is okay for the log.

Reply:
Ack, can resolve the comment.

Reply:
> Just missed why after all `RemoveImageCommand::setSucceeded()` is needed if it only calls it's `super.setSucceeded()`...

+1
@mkemel can you explain that part?